### PR TITLE
Update catalog-info.yaml

### PR DESCRIPTION
**Pull request by jenreiher-bcgov**
Add title and update description to make it more user-friendly when viewed in the techdocs page

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -2,7 +2,8 @@ apiVersion: backstage.io/v1alpha1
 kind: Component
 metadata:
   name: "platform-developer-docs"
-  description: "A guide to application development for the Government of the Province of British Columbia"
+  title: "Private cloud development guide"
+  description: "Learn how to deploy applications on the private cloud OpenShift environment"
```
Comment (Pilargit12) [inline catalog-info.yaml:6]: Learn how to build, deploy, maintain, and retire applications on OpenShift and explore additional technical guidelines for a deeper understanding.

Comment (Pilargit12) [inline catalog-info.yaml:6]: description: "Learn how to build, deploy, maintain, and retire applications on OpenShift and explore additional technical guidelines for a deeper understanding."

Comment (jenreiher-bcgov) [reply catalog-info.yaml]: @Pilargit12 done! I removed the . because the other descriptions don't have one

**Review by Pilargit12 — APPROVED** [review]
Thank you! Looks great!

**Pull request merged**